src/components/Proof.js

```diff
@@ -22,29 +22,9 @@ export default {
 			return this.$store.state.contractReady;
 		},
 		proof : function() {
-			let hash = this.$route.params.id;
-			let data = {
-				image: this.$store.state.apiBaseUrl + '/uploads/' + hash,
-				title: '',
-				fileSha256: hash,
-				created: null,
-				ipfsHash: '',
-				contract: '',
-				owner: '',
-				block: ''
-			};
 
-			if (this.rawProof) {
-				data.contract = this.$store.state.contractAddress;
-				data.owner = this.rawProof[0];
-				data.created = this.rawProof[1];
-				data.block = this.rawProof[2];
-				data.title = this.rawProof[3];
-				data.ipfsHash = this.rawProof[4];
-				data.fileSha256 = this.rawProof[5];
-			}
-
-			return data;
+      let proof = this.$store.state.proofs.find((el)=>{return el.fileSha256 = this.$route.params.id;});
```

Comment:
this will only work if you display a proof you proofed yourself

Reply:
I see what u mean. IMO then `this.$route.params.id` should be the ipfs has instead of the sha. That way I could use it to query ipfs for the remote file. otherwise, wouldn't we have to do a lookup-by-hash using the contract, would cost money?

Reply:
the lookup doesnt cost anything since it's a "constant" function, it doesnt alter the state of the contract. its already happening in the "getProof" function in the Proof.js.
the hash basically is the id of of the proof. its always present. We allow users to proof files without uploading them to ipfs, thats why ipfsHash may not always be present

Reply:
got you.

Reply:
I reverted this to last commit on master and fetch the image from ipfs in getProof() and store it in a data prop just like rawProof. What do you think about using https://alligator.io/vuejs/async-computed-properties/. For now this way works but it's not totally deterministic and would probably start acting up at scale.